### DISCUSSION TITLE
HOGWARTS: Cannot figure out how to toggle form in JS instead of hard code.

### DISCUSSION THREAD

**Discussion by BenJarrett** [nss-evening-cohort-14/evening-client-side#82] [Help Me!] [answered]
Ben Jarrett

I have searched and read multiple tickets. I have found similar tickets, but none that directly correlate to this issue. 
I could easily not be putting two and two together on a previous ticket, but I have looked and read through many. 
#65 
#61 
#70 
#60 

In my earlier ticket, I was having trouble figuring out how to have my form not show up on App Load. 
Dr. T then informed me that it was because I had the onclick hard-coded into my html and to do this in my JS. 
This worked, and my form was not showing up on App Load. 

I cannot figure out how to exactly toggle the form in JS, instead of in the HTML. 

I have no errors.

https://medium.com/swlh/creating-hidden-forms-using-javascript-286356a13dc4
https://stackoverflow.com/questions/11226489/show-hide-forms-using-buttons-and-javascript
https://www.washington.edu/accesscomputing/webd2/student/unit5/module2/lesson5.html
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/hidden
https://css-tricks.com/places-its-tempting-to-use-display-none-but-dont/

So I have tried quite a few things. 

I tried to add a method chain event listener into my buttonEvents function
  ```
document.querySelector('#sortBtn').addEventListener('click', handleButtonClick);
  document.querySelector('#toggle').addEventListener('click', handleButtonClick);
}
```

I have tried declaring a new function that adds an event listener outside of this function. 

I tried adding this to 

```
 function toggleHide() {
  const hidesBtn = document.querySelector("#toggle");
  const unhide = document.querySelector('#toggle');
  if (hidesBtn.style.display === "none") {
    hidesBtn.style.display = "block";
  } else {
    hidesBtn.style.display = "none";
  }
}
```
https://github.com/BenJarrett/sorting-hat/tree/hide-sorting

**Comment by TrinityChristiana** [comment]
I don't see your toggleHide function being called anywhere. Is that intentional?

**Reply by BenJarrett** [reply]
It is not intentional. Is that something I should intialize in 


``` const init = () => {
  buttonEvents();
  
};

**Reply by BenJarrett** [reply]
This is inside my ``` const init = () => {
}

**Reply by TrinityChristiana** [reply]
When do you want that function to be called? Is it here: https://github.com/BenJarrett/sorting-hat/blob/hide-sorting/main.js#L56

Looks like on toggle click it's running this callback: handleButtonClick

**Reply by BenJarrett** [reply]
Should I be calling it as a CB instead of the handleButtonClick?

**Reply by TrinityChristiana** [reply]
Yes.

**Reply by BenJarrett** [reply]
Okay, just tried that and still it isn't toggling. There must be something else wrong with my code

**Comment by BenJarrett** [comment, marked as answer]
Okay... Wow. 
Can't believe I missed this. 

Thank you to Trinity, Aja, and Dr T. on this one. 

So I had two issues here. 

First, I need to add an id to the jumbotron div. 
``` 
<p class="lead">
        <a class="btn btn-primary btn-lg" href="#" role="button" id="pleaseToggle">Let's get to sorting!</a>
      </p> 
``` 

Next, I needed to target this id and use my toggleHide function as the cb for the event listener in my buttonEvents function. 
``` 
const buttonEvents = () => {
  document.querySelector('#sortBtn').addEventListener('click', handleButtonClick);
  document.querySelector('#pleaseToggle').addEventListener('click', toggleHide);

}
```
